specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/virtualMachine.json

```diff
@@ -1891,6 +1891,331 @@
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Delete an application from a vm": {
+            "$ref": "./examples/virtualMachineExamples/VirtualMachineApplications_Delete.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "VirtualMachineApplications"
+        ],
+        "operationId": "VirtualMachineApplications_Put",
+        "description": "The operation to add an application on a Virtual Machine.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "vmName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the virtual machine."
+          },
+          {
+            "name": "applicationName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the application to delete."
+          },
+          {
+            "name": "application",
+            "in": "body",
+            "required": true,
+            "description": "The definition of the VMApplication to add the virtual machine",
+            "schema": {
+              "$ref": "./computeRPCommon.json#/definitions/VMApplicationProxyResourceWithInstanceView"
+            }
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Update, OK",
+            "schema": {
+              "$ref": "./computeRPCommon.json#/definitions/VMApplicationProxyResourceWithInstanceView"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "./computeRPCommon.json#/definitions/VMApplicationProxyResourceWithInstanceView"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common-types/v1/common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "VirtualMachineApplication_Put": {
+            "$ref": "./examples/virtualMachineExamples/VirtualMachineApplications_CreateOrUpdate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/vmapplications/{applicationName}/instanceView": {
```

Comment:
>instanceView [](http://example.com/codeflow?start=158&length=12)

plural?

Reply:
did u intend this to be a point get? if so the api path is wrong

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/vmapplications/{applicationName}/instanceViews/default

is more appropriate. odd number of segments indicate the resource collection. for singular objects you need to use even segmented paths

Reply:
there is precedence on other Microsoft.Compute resources, and I am trying to adhere to the same pattern.
 
https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/instance-view?view=rest-compute-2023-07-01&tabs=HTTP

Reply:
Approving because of this being an existing pattern